src/hooks/useMap.js

```diff
@@ -1,44 +1,139 @@
-  const [topic, setTopic] = useState(null);
+  const [topicId, setTopicId] = useState(null);
 
   useEffect(() => {
+    const loadSelectedTarget = () => {
+      if (!selectedTargetId) return;
+
+      const { target } = targets.find(
+        ({ target }) => target.id == selectedTargetId
+      );
+
+      if (target.title !== title) setTitle(target.title);
+      if (topicId !== target.topic.id) setTopicId(target.topic.id);
+
+      const newMapProperties = {
+        ...mapProperties,
+        selectedRadius: target.radius,
+        selectedLocation: {
+          lat: target.latitude,
+          lng: target.longitude,
+        },
+      };
+
+      if (!isEqual(newMapProperties, mapProperties))
+        setMapProperties(newMapProperties);
+    };
+
+    loadSelectedTarget();
+  }, [
+    selectedTargetId,
+    mapProperties,
+    setMapProperties,
+    targets,
+    title,
+    topicId,
+  ]);
+
+  useEffect(() => {
+    const fetchTopics = async () => {
+      const receivedTopics = await TopicsService.getTopics();
+      const formattedTopics = receivedTopics.map(({ topic }) => ({
+        key: topic.id,
+        value: topic.id,
+        label: topic.label,
+      }));
+
+      if (!isEqual(formattedTopics, topicsList)) setTopicsList(formattedTopics);
+    };
+
     fetchTopics();
-  }, []);
-
-  const fetchTopics = async () => {
-    const receivedTopics = await TopicsService.getTopics();
-    const formattedTopics = receivedTopics.map(({ topic }) => ({
-      key: topic.id,
-      value: topic.id,
-      label: topic.label,
-    }));
-    setTopicsList(formattedTopics);
+  }, [topicsList]);
+
+  useEffect(() => {
+    const fetchCurrentLocation = () => {
+      navigator.geolocation.getCurrentPosition((position) => {
+        if (mapProperties.location) return;
+
+        setMapProperties({
+          ...mapProperties,
+          location: {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude,
+          },
+        });
+      });
+    };
+
+    const fetchTargets = async () => {
+      const response = await TargetsService.getTargets();
+      const targetsIds = targets.map(({ id }) => id).sort();
+      const responseIds = response.map(({ id }) => id).sort();
+
+      if (!isEqual(targetsIds, responseIds)) setTargets(response);
+    };
+
+    fetchCurrentLocation();
+    fetchTargets();
+  }, [mapProperties, targets, setMapProperties, setTargets]);
+
+  const handleMapClick = (e) => {
+    setMapProperties({
+      ...mapProperties,
+      selectedLocation: { lat: e.latLng.lat(), lng: e.latLng.lng() },
+      selectedTargetId: null,
+    });
+  };
+
+  const handleTargetClick = (targetId) => {
+    setMapProperties({
+      ...mapProperties,
+      selectedTargetId: targetId,
+      selectedLocation: null,
+    });
+  };
+
+  const isSelectedTargetStored = () => {
+    return (
+      mapProperties.selectedLocation != null &&
+      mapProperties.selectedTargetId != null
+    );
   };
 
   return {
```

Comment:
this can be just a returned variable 
```suggestion
  const isSelectedTargetStored = () => {
    return (
      mapProperties.selectedLocation != null &&
      mapProperties.selectedTargetId != null
    );
  };

  return {
    isSelectedTarget: !!mapProperties.selectedLocation && !!mapProperties.selectedTargetId
```